Makefile

```diff
@@ -81,12 +98,32 @@ distclean:
 	-rm -rf deps/icu
 	-rm -rf deps/icu4c*.tgz deps/icu4c*.zip deps/icu-tmp
 	-rm -f $(BINARYTAR).* $(TARBALL).*
+	-rm -rf deps/v8/testing/gmock
+	-rm -rf deps/v8/testing/gtest
 
 check: test
 
 cctest: all
 	@out/$(BUILDTYPE)/$@
 
+v8:
+	tools/make-v8.sh v8
+ifneq (,$(filter $(DESTCPU),x86))
+	+$(MAKE) -C deps/v8 $(V8_BUILD_OPTIONS)
+else
+ifneq (,$(filter $(ARCH),x86))
+	+$(MAKE) -C deps/v8 $(V8_BUILD_OPTIONS)
+else
+ifeq ($(ARCH)$(DESTCPU),)
+	+$(MAKE) -C deps/v8 $(V8_BUILD_OPTIONS)
+else
+	+$(MAKE) -C deps/v8 $(ARCH) $(V8_BUILD_OPTIONS)
+endif
+	+$(MAKE) -C deps/v8 $(ARCH) $(V8_BUILD_OPTIONS)
+endif
+	+$(MAKE) -C deps/v8 $(ARCH) $(V8_BUILD_OPTIONS)
```

Comment:
I'd stick with ARCH and drop the logic for DESTCPU.  Less code, easier to comprehend.

Also, if you're not sure why the plus signs are there, can you please remove them?